[Docs]: i18n - How-to translate plugin (#1960)
grafana/plugin-tools: docusaurus/docs/how-to-guides/panel-plugins/plugin-internationalization.md

```

### Determine the text to translate

After you've configured your plugin for translation you can proceed to mark up the language strings you want to translate. Each translatable string is assigned an unique key that ends up in each translation file under `locales/<locale>/<plugin id>.json`. 

For example:

```tsx title="ColorEditor.tsx"
import { t } from '@grafana/i18n';

export function ColorEditor(props: any) {
  const styles = getStyles(config.theme);
  const defaultValue = t('colorEditor.defaultValue', 'Pick Color');
  let prefix: React.ReactNode = null;
  let suffix: React.ReactNode = null;
  if (props.value) {
    suffix = <Icon className={styles.trashIcon} name="trash-alt" onClick={() => props.onChange(undefined)} />;
  }

  prefix = (
    <div className={styles.inputPrefix}>
      <div className={styles.colorPicker}>
        <ColorPicker
          color={props.value || config.theme.colors.panelBg}
          onChange={props.onChange}
          enableNamedColors={true}
        />
      </div>
    </div>
  );

  return (
    <div>
      <Input type="text" value={props.value || defaultValue} prefix={prefix} suffix={suffix} readOnly={true} />
    </div>
  );
}  

> ```

Author: Do you have an example handy I can use?

export function ColorEditor(props: any) {
  const styles = getStyles(config.theme);
  const defaultValue = t('colorEditor.defaultValue', 'Pick Color');
  let prefix: React.ReactNode = null;
  let suffix: React.ReactNode = null;
  if (props.value) {
    suffix = <Icon className={styles.trashIcon} name="trash-alt" onClick={() => props.onChange(undefined)} />;
  }

  prefix = (
    <div className={styles.inputPrefix}>
      <div className={styles.colorPicker}>
        <ColorPicker
          color={props.value || config.theme.colors.panelBg}
          onChange={props.onChange}
          enableNamedColors={true}
        />
      </div>
    </div>
  );

  return (
    <div>
      <Input type="text" value={props.value || defaultValue} prefix={prefix} suffix={suffix} readOnly={true} />
    </div>
  );
}  
```

You can aslo modify `options.ts` and `RenderTime.tsx`.

### Extract marked up tags automatically

Use the `i18next` [parser](https://github.com/i18next/i18next-parser#readme) to sweep all input files, extract tagged `i18n` keys, and write them into the `<plugin id>.json` output file, located under `locales/en-US`: 

```shell
yarn add @grafana/i18n@latest

> ```

Author: Done!

Next, include `i18next` as an external in your webpack configuration:

```shell
yarn add -D i18next-parser

> ```

Reviewer: I think this can be removed

```

### Define the languages and Grafana dependencies

Set up the translation languages for your plugin. 

To do so, insert a `languages` section into the `plugin.json` file of the plugin. For example, if you want to add US English and Brazilian Portuguese:

```json title="plugin.json"
"dependencies": {
    "grafanaDependency": ">=11.0.0", // @grafana/i18n works from version 11.0.0 and higher for en-US translations 
    "plugins": []
  }
"languages": ["en-US", "pt-BR"] // the languages that the plugin supports
```

### Extend your plugin configuration to include translation

Add the latest version of the `@grafana/i18n` translation package:

```shell npm2yarn
npm install @grafana/i18n@latest
```

Next, mark `i18next` as an external in your Webpack configuration. See how in [Extend default configurations](https://grafana.com/developers/plugin-tools/how-to-guides/extend-configurations).

```ts title="webpack.config.ts"
const config = async (env: Record<string, unknown>): Promise<Configuration> => {
  const baseConfig = await grafanaConfig(env);
  const externals = baseConfig.externals as string[];

  return merge(baseConfig, { externals: [...externals, 'i18next'] });

> };

Reviewer: Opened a PR to fix the existing docs: https://github.com/grafana/plugin-tools/pull/1966

```

### Include translation in `module.ts` 

Add plugin translation to `module.ts`: 

```ts title="module.ts"
import { initPluginTranslations } from '@grafana/i18n';
import pluginJson from 'plugin.json';

await initPluginTranslations(pluginJson.id);

> ```

Author: Yay!

---
id: plugin-internationalization
title: How to add translations to your plugin
description: Translate your plugins
keywords:
  - grafana
  - plugins
  - plugin
  - frontend
  - tooling
  - configuration
  - translation
  - localization
  - internationalization
  - webpack
---

# How to add translations to your plugin

The Grafana UI is available in [several languages](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-language). By default, plugins are available in English only and are not translated when you change your language settings. 

> Grafana uses the [i18next internationalization framework](https://www.i18next.com/) for translation purposes. If you want your plugin to be translatable to other languages you need to perform the changes described in this document.  

Reviewer: Sound good to me 👍

* Basic knowledge of Grafana panel plugin development
* Basic understanding of the [i18next internationalization framework](https://www.i18next.com/)

## Overview of the files affected by translation

If you create your plugin running the `create-plugin` scaffolding tool, enabling plugin translation involves updating the following files:

- `docker-compose.yaml` 
- `plugin.json`
- `module.ts`
- `webpack.config.ts`
- `package.json`

By the end of the translation process you'll have a file structure like this:

```
myorg-myplugin-plugintype/
├── pkg/
│   ├── main.go
│   └── plugin/
├── src/
│   ├── locales
│      ├── en-US
│         └──--.json 
│      ├── pt-BR
│         └──--.json 
│   ├── module.ts
│   ├── plugin.json
└── tests/
├── docker-compose.yaml
├── webconfig.ts
├── package.json
```

## Set up your plugin for translation

Follow these steps to update your plugin and set it up for translation. While this example is based on a panel plugin, the process is the same for data source and app plugins.

### Enable translation in your Grafana instance 

To translate your plugin you need to enable the feature toggle `localizationForPlugins` on your Grafana instance.

> 

Author: Good catch! I'll fix this.

```json title="plugin.json"
"dependencies": {
    "grafanaDependency": ">=12.1.0", // @grafana/i18n works from version 11.0.0 and higher for en-US translations 
    "plugins": []
  }
"languages": ["en-US", "pt-BR"] // the languages that the plugin supports
```

### Extend your plugin configuration to include translation

Install the latest version of the `@grafana/i18n` translation package:

```shell npm2yarn
npm install @grafana/i18n@latest
```

Next, mark `i18next` as an external in your Webpack configuration. See how in [Extend default configurations](https://grafana.com/developers/plugin-tools/how-to-guides/extend-configurations).

```ts title="webpack.config.ts"
import type { Configuration } from 'webpack';
import { merge } from 'webpack-merge';
import grafanaConfig, { Env } from './.config/webpack/webpack.config';

const config = async (env: Env): Promise<Configuration> => {
  const baseConfig = await grafanaConfig(env);
  const externals = baseConfig.externals as string[];
  return merge(baseConfig, { externals: [...externals, 'i18next'] });
};

export default config;

> ```

Author: This is a good call, adding it!

### Initialize translations in `module.ts` 

Add plugin translation to `module.ts`: 

```ts title="module.ts"
import { initPluginTranslations } from '@grafana/i18n';
import pluginJson from 'plugin.json';

await initPluginTranslations(pluginJson.id);
```

## Determine the text to translate

After you've configured your plugin for translation you can proceed to mark up the language strings you want to translate. Each translatable string is assigned an unique key that ends up in each translation file under `locales/<locale>/<plugin id>.json`. 

For example:

> 

Reviewer: Updated 👍

+import { Trans } from '@grafana/i18n';
 
 interface Props extends PanelProps<SimpleOptions> {}
 
@@ -60,9 +61,15 @@ export const SimplePanel: React.FC<Props> = ({ options, data, width, height, fie
 
       <div className={styles.textBox}>
         {options.showSeriesCount && (
-          <div data-testid="simple-panel-series-counter">Number of series: {data.series.length}</div>
+          <div data-testid="simple-panel-series-counter">
+            <Trans i18nKey="components.simpePanel.options.showSeriesCount">
+              Number of series: {{ numberOfSeries: data.series.length }}
+            </Trans>
+          </div>
         )}
-        <div>Text option value: {options.text}</div>
+        <Trans i18nKey="components.simpePanel.options.textOptionValue">
+          Text option value: {{ optionValue: options.text }}
+        </Trans>
       </div>
     </div>
   );
```

## Obtain the translated text

Use the `i18next` [parser](https://github.com/i18next/i18next-parser#readme) and [exta] to sweep all input files, extract tagged `i18n` keys, and save the translations. 

### Parse for translations

Install the `i18next` parser:

```shell npm2yarn
npm install i18next-parser
```

> Next, create a configuration file `src/locales/i18next-parser.config.js` and configure it so the parser sweeps your plugin and extracts the translations into the `locales/[$LOCALE]/[your-plugin].json`:

Author: Done!

> Next, create a configuration file `src/locales/i18next-parser.config.js` and configure it so the parser sweeps your plugin and extracts the translations into the `locales/[$LOCALE]/[your-plugin].json`:

Reviewer: Awesome 🙌

---
id: plugin-internationalization
title: How to add translations to your plugin
description: Translate your plugins
keywords:
  - grafana
  - plugins
  - plugin
  - frontend
  - tooling
  - configuration
  - translation
  - localization
  - internationalization
  - webpack
---

# How to add translations to your plugin

The Grafana UI is available in [several languages](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-language). By default, plugins are available in English only and are not translated when you change your language settings. 

Grafana uses the [i18next internationalization framework](https://www.i18next.com/) for translation purposes. If you want your plugin to be translatable to other languages you need to perform the changes described in this document.  

## Before you begin

:::info
Available in Grafana >=v11.0.0 (US English) and Grafana >=v12.1.0 for the rest of languages.
:::

The following is recommended:

* Basic knowledge of Grafana panel plugin development
* Basic understanding of the [i18next internationalization framework](https://www.i18next.com/)

## Overview of the files affected by translation

> If you create your plugin running the `create-plugin` scaffolding tool you'll get the following folder layout and key file:

Reviewer: Not sure what to do here, but should we mention that this `how to example` is for a scaffold panel plugin but its the same process for data source plugins and app plugins?

Reviewer: Maybe we should only list the files that are affected by translation as the heading says?

<orgName>-<pluginName>-<pluginType>
├── src
│   ├── locales
│   │   ├── en-US
│   │   │   └── <orgName>-<pluginName>-<pluginType>.json
│   │   ├── pt-BR
│   │   │   └── <orgName>-<pluginName>-<pluginType>.json
│   │   └── i18next-parser.config.js
│   ├── module.ts
│   └── plugin.json
├── docker-compose.yaml
├── web.config.ts
└── package.json

```
myorg-myplugin-plugintype/
├── pkg/
│   ├── main.go

> │   └── plugin/

Reviewer: Maybe just remove all backend related code to reduce the noise?

Author: Here at the beginning of the doc I'd keep a simplified version of the brand-new post-scaffolding structure. And then we can mention the files that will be created along the process. WDYT?

Reviewer: Yes, that sounds like a plan 👍  But maybe remove the backend related files at least?

├── src/
│   ├── module.ts
│   ├── plugin.json
└── tests/
├── CHANGELOG.md
├── docker-compose.yaml

> ├── go.mod

Reviewer: Maybe just remove all backend related code to reduce the noise?

├── package.json
├── LICENSE
├── Magefile.go

> ├── README.md

Reviewer: Maybe just remove all backend and other related code to reduce the noise?

```

:::note

Since `create-plugin` is constantly being improved your folder structure could look slightly different.

:::

Enabling plugin translation involves updating the following files:

- `docker-compose.yaml` 
- `plugin.json`
- `module.ts`
- `webpack.config.ts`
- `package.json`

## Set up your plugin for translation

Follow these steps to update your plugin and set it up for translation:

### Enable translation in your Grafana instance 

To translate your plugin you need to enable the feature toggle `localizationForPlugins` on your Grafana instance.

To do so, update `docker-compose.yaml` in your plugin:

```yaml title="docker-compose.yaml"
services:
  grafana:
    extends:
      file: .config/docker-compose-base.yaml
      service: grafana
    environment:
      GF_FEATURE_TOGGLES_ENABLE: localizationForPlugins
```

### Define the languages and Grafana dependencies

Set up the translation languages for your plugin. 

> To do so, insert a `languages` section into the `plugin.json` file of the plugin. For example, if you want to add US English and Brazilian Portuguese:

Reviewer: I think we need to emphasize the `grafanaDependency` change again, super important because this is probably a breaking change for many plugins.

```json title="plugin.json"
"dependencies": {
    "grafanaDependency": ">=11.0.0", // @grafana/i18n works from version 11.0.0 and higher for en-US translations 
    "plugins": []
  }
"languages": ["en-US", "pt-BR"] // the languages that the plugin supports
```

### Extend your plugin configuration to include translation

Add the latest version of the `@grafana/i18n` translation package:

```shell npm2yarn
npm install @grafana/i18n@latest
```

Next, mark `i18next` as an external in your Webpack configuration. See how in [Extend default configurations](https://grafana.com/developers/plugin-tools/how-to-guides/extend-configurations).

```ts title="webpack.config.ts"
import type { Configuration } from 'webpack';
import { merge } from 'webpack-merge';
import grafanaConfig, { Env } from './.config/webpack/webpack.config';

const config = async (env: Env): Promise<Configuration> => {
  const baseConfig = await grafanaConfig(env);
  const externals = baseConfig.externals as string[];
  return merge(baseConfig, { externals: [...externals, 'i18next'] });
};

export default config;
```

> ### Include translation in `module.ts` 

Reviewer: ```suggestion
### Initialize translations in `module.ts` 
```

    return <PanelDataErrorView fieldConfig={fieldConfig} panelId={id} data={data} needsStringField />;
  }

  return (
    <div
      className={cx(
        styles.wrapper,
        css`
          width: ${width}px;
          height: ${height}px;
        `
      )}
    >
      <svg
        className={styles.svg}
        width={width}
        height={height}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
        viewBox={`-${width / 2} -${height / 2} ${width} ${height}`}
      >
        <g>
          <circle data-testid="simple-panel-circle" style={{ fill: theme.colors.primary.main }} r={100} />
        </g>
      </svg>

      <div className={styles.textBox}>
        {options.showSeriesCount && (
          <div data-testid="simple-panel-series-counter">
            <Trans i18nKey="components.simpePanel.options.showSeriesCount">
              Number of series: {{ numberOfSeries: data.series.length }}
            </Trans>
          </div>
        )}
        <Trans i18nKey="components.simpePanel.options.textOptionValue">
          Text option value: {{ optionValue: options.text }}
        </Trans>
      </div>
    </div>
  );
};

> ```

Reviewer: ```diff
import { SimpleOptions } from 'types';
 import { css, cx } from '@emotion/css';
 import { useStyles2, useTheme2 } from '@grafana/ui';
 import { PanelDataErrorView } from '@grafana/runtime';
+import { Trans } from '@grafana/i18n';
 
 interface Props extends PanelProps<SimpleOptions> {}
 
@@ -60,9 +61,15 @@ export const SimplePanel: React.FC<Props> = ({ options, data, width, height, fie
 
       <div className={styles.textBox}>
         {options.showSeriesCount && (
-          <div data-testid="simple-panel-series-counter">Number of series: {data.series.length}</div>
+          <div data-testid="simple-panel-series-counter">
+            <Trans i18nKey="components.simpePanel.options.showSeriesCount">
+              Number of series: {{ numberOfSeries: data.series.length }}
+            </Trans>
+          </div>
         )}
-        <div>Text option value: {options.text}</div>
+        <Trans i18nKey="components.simpePanel.options.textOptionValue">
+          Text option value: {{ optionValue: options.text }}
+        </Trans>
       </div>
     </div>
   );
```

Reviewer: Maybe we should have this sort of diff before and after the added translations, instead of the whole file? You can use the copy button to copy the exact contents of the comment above 👍

## Extract translated text automatically

Use the `i18next` [parser](https://github.com/i18next/i18next-parser#readme) to sweep all input files, extract tagged `i18n` keys, and save the translations. 

To do so, install the parser:

```shell npm2yarn
npm install i18next-parser
```

> Next, create a configuration file (for example, `i18next-parser.config.js`) and configure it so the parser sweeps your plugin and extracts the translations into the `locales/[$LOCALE]/[your-plugin].json`:

Reviewer: Small nit, should we mention where the reader should create this file by adding the relative path?
```suggestion
Next, create a configuration file `src/locales/i18next-parser.config.js` and configure it so the parser sweeps your plugin and extracts the translations into the `locales/[$LOCALE]/[your-plugin].json`:
```

```js title="i18next-parser.config.js"
const pluginJson = require('../plugin.json');

module.exports = {
  locales: ['en-US'], // Only en-US  is updated - Crowdin will PR with other languages
  sort: true,
  createOldCatalogs: false,
  failOnWarnings: true,
  verbose: false,
  resetDefaultValueLocale: 'en-US', // Updates extracted values when they change in code
  defaultNamespace: pluginJson.id,
  input: ['../**/*.{tsx,ts}'],
  output: 'src/locales/$LOCALE/$NAMESPACE.json',
};
```

Finally, add `i18n-extract` to `package.json`:

```json title="package.json"
  "scripts": {
    "i18n-extract": "i18next --config src/locales/i18next-parser.config.js",
  },    
```

> 

Reviewer: Maybe add an example on how to run the command and a describe how running that command will create the translation file.

```shell npm2yarn
npm run i18n-extract
```

Reviewer: and maybe add an example of how that file would look like after the run? Let me know if this is to much information for the reader

```json
{
  "components": {
    "simpePanel": {
      "options": {
        "showSeriesCount": "Number of series: {{numberOfSeries}}",
        "textOptionValue": "Text option value: {{optionValue}}"
      }
    }
  }
}
```

---
id: plugin-internationalization
title: How to add translations to your plugin
description: Translate your plugins
keywords:
  - grafana
  - plugins
  - plugin
  - frontend
  - tooling
  - configuration
  - translation
  - localization
  - internationalization

>   - webpack

Reviewer 2: nit: dunno what weight keywords bring these days but this doc doesn't feel too webpack specific to me so would consider dropping from this list.

---

# How to add translations to your plugin

The Grafana UI is available in [several languages](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-language). By default, plugins are available in English only and are not translated when you change your language settings. 

Grafana uses the [i18next internationalization framework](https://www.i18next.com/) for translation purposes. If you want your plugin to be translatable to other languages you need to perform the changes described in this document.  

## Before you begin

:::info

> Available in Grafana >=v11.0.0 (US English) and Grafana >=v12.1.0 for the rest of languages.

Reviewer 2: I'm a little confused by this sentence. Does it mean plugins that include translations will continue to work in Grafana >= 11.0.0 but the translations will only work in >=12.1.0?

Author: I thought it was pretty clear, but I can reword it if it's confusing...

:::

The following is recommended:

> * Basic knowledge of Grafana panel plugin development

Reviewer 2: Later in this doc it mentions translations can be applied to any plugin type. If that's correct I'd consider the following:

```suggestion
* Basic knowledge of Grafana plugin development
```

* Basic understanding of the [`t` function](https://www.i18next.com/overview/api#t)
* Basic understanding of the [`Trans` component](https://react.i18next.com/latest/trans-component)

## Overview of the files affected by translation

If you create your plugin running the `create-plugin` scaffolding tool, enabling plugin translation involves updating the following files:

- `docker-compose.yaml` 
- `plugin.json`
- `module.ts`
- `webpack.config.ts`
- `package.json`

By the end of the translation process you'll have a file structure like this:

```
myorg-myplugin-plugintype/
├── pkg/
│   ├── main.go
│   └── plugin/
├── src/
│   ├── locales
│      ├── en-US

> │         └──--.json 

Reviewer 2: Don't these json files have names based on the plugin-id?
```suggestion
│         └──myorg-myplugin-plugintype.json 
```

Reviewer: Good catch 🎉

│      ├── pt-BR
│         └──--.json 
│   ├── module.ts
│   ├── plugin.json
└── tests/
├── docker-compose.yaml
├── webconfig.ts
├── package.json
```

## Set up your plugin for translation

> Follow these steps to update your plugin and set it up for translation. While this example is based on a panel plugin, the process is the same for data source and app plugins.

Reviewer 2: I would consider moving/merging this sentence with the `How to add translations to your plugin` section at the top of the page and moving this doc up a level to `how-to-guides`. 

It living in `/how-to-guides/panel-plugins` makes me believe only panel plugins can be translated which I don't believe is the case.

Reviewer: I totally agree with you Jack, I missed this in my review 🙈

```json title="plugin.json"
"dependencies": {
    "grafanaDependency": ">=12.1.0", // @grafana/i18n works from version 11.0.0 and higher for en-US translations 
    "plugins": []
  },
"languages": ["en-US", "pt-BR"] // the languages that the plugin supports
```

### Extend your plugin configuration to include translation

Install the latest version of the `@grafana/i18n` translation package:

```shell npm2yarn
npm install @grafana/i18n@latest
```

Next, mark `i18next` as an external in your Webpack configuration. See how in [Extend default configurations](https://grafana.com/developers/plugin-tools/how-to-guides/extend-configurations).

```ts title="webpack.config.ts"
import type { Configuration } from 'webpack';
import { merge } from 'webpack-merge';
import grafanaConfig, { Env } from './.config/webpack/webpack.config';

const config = async (env: Env): Promise<Configuration> => {
  const baseConfig = await grafanaConfig(env);
  const externals = baseConfig.externals as string[];
  return merge(baseConfig, { externals: [...externals, 'i18next'] });

> };

Reviewer 2: Maybe it's just me but I'm not sure we want to ask people to extend the externals in their webpack configs. It might be nicer to provide a featureFlag (`useTranslations`) within create-plugin itself which would make the necessary changes for them. This is how we approached react-router v6 update and bundling grafana/ui. Then they can run `npx @grafana/create-plugin update --feature-flags=useTranslations` and the cli will do the necessary changes.

Author: Let's leave as it is for the moment. We'll remove it when `i18next` becomes an external in `.config/webpack.config.ts`

:::caution
Remember to change the path to `webpack.config.ts` in `package.json`.

> :::

Reviewer: Not sure we can use this caution box here, do you know?

Reviewer 2: Nope - we cannot use admonitions within code blocks. Needs to be moved below the closing ``` for it render correctly.

Author: Why not? What'd be the issue?

         {options.showSeriesCount && (
-          <div data-testid="simple-panel-series-counter">Number of series: {data.series.length}</div>
+          <div data-testid="simple-panel-series-counter">
+            <Trans i18nKey="components.simpePanel.options.showSeriesCount">
+              Number of series: {{ numberOfSeries: data.series.length }}
+            </Trans>
+          </div>
         )}
-        <div>Text option value: {options.text}</div>
+        <Trans i18nKey="components.simpePanel.options.textOptionValue">
+          Text option value: {{ optionValue: options.text }}
+        </Trans>
       </div>
     </div>
   );
```

## Obtain the translated text

Use the `i18next` [parser](https://github.com/i18next/i18next-parser#readme) and [exta] to sweep all input files, extract tagged `i18n` keys, and save the translations. 

### Parse for translations

Install the `i18next` parser:

```shell npm2yarn
npm install i18next-parser
```

Next, create a configuration file `src/locales/i18next-parser.config.js` and configure it so the parser sweeps your plugin and extracts the translations into the `locales/[$LOCALE]/[your-plugin].json`:

:::danger

> The path `locales/[$LOCALE]/[your-plugin].json` is mandatory. If you modify it translations won't work.

Reviewer 2: ```suggestion
The path `locales/[$LOCALE]/[your-plugin-id].json` is mandatory. If you modify it translations won't work.
```

:::

```js title="i18next-parser.config.js"
const pluginJson = require('../plugin.json');

module.exports = {

>   locales: ['en-US'], // Only en-US  is updated - Crowdin will PR with other languages

Reviewer 2: Not sure this comment about Crowdin should be in this file? Isn't it only Grafana that receives PRs with crowdin updates?

Author: Done!

  sort: true,
  createOldCatalogs: false,
  failOnWarnings: true,
  verbose: false,
  resetDefaultValueLocale: 'en-US', // Updates extracted values when they change in code
  defaultNamespace: pluginJson.id,
  input: ['../**/*.{tsx,ts}'],
  output: 'src/locales/$LOCALE/$NAMESPACE.json',
};
```

### Obtain your translation file

Add the translation script `i18n-extract` to `package.json`:

```json title="package.json"
  "scripts": {
    "i18n-extract": "i18next --config src/locales/i18next-parser.config.js",
  },    
```

Run the script to translate the files:

```shell npm2yarn
npm run i18n-extract
```

The translation file will look similar to this:

```
{
  "components": {

>     "simpePanel": {

Reviewer 2: ```suggestion
    "simplePanel": {
```

Reviewer: Great catch Jack 🙌 , that needs to change in the example code as well and the json output too
